pennylane/plugins/default_gaussian.py

```diff
@@ -793,12 +795,12 @@ def expval(self, observable, wires, par):
         return ev
 
     def var(self, observable, wires, par):
-        # get indices of wires on this device's register
-        registers = self.wire_map(wires)
 
-        mu, cov = self.reduced_state(registers)
+        wire_indices = self.indices_on_register(wires)
+
+        mu, cov = self.reduced_state(wire_indices)
         _, var = self._observable_map[observable](
-            mu, cov, registers, par, hbar=self.hbar, total_wires=self.num_wires
+            mu, cov, wire_indices, par, self.num_wires, self.register, hbar=self.hbar
```

Comment:
Yes. Again, I would propose not to meddle with the device and focus on this already gigantic PR :) Let's put a refactor on the roadmap!

Reply:
I'm going to disagree here. We're already meddled with the device by changing the signature of several functions unnecessarily. If it's unavoidable to have to have to make changes, I'd rather the changes were easier to follow/justify (e.g., having a single conditional for the one special case) rather than try to force changes confusingly to several functions in order to not "break the abstraction"

Reply:
Convinced, I will implement the if statement!